File: app/src/androidTest/java/com/teamdecano/cryptocoin/root/RootInteractorTest.java
package com.teamdecano.cryptocoin.root;

import com.uber.rib.core.RibTestBasePlaceholder;
import com.uber.rib.core.InteractorHelper;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

public class RootInteractorTest extends RibTestBasePlaceholder {

    @Mock RootInteractor.RootPresenter presenter;
    @Mock RootRouter router;

    private RootInteractor interactor;

    @Before
    public void setup() {
        MockitoAnnotations.initMocks(this);

        interactor = TestRootInteractor.create(presenter);
    }

    /**
     * TODO: Delete this example and add real tests.
     */
    @Test
    public void anExampleTest_withSomeConditions_shouldPass() {
        // Use InteractorHelper to drive your interactor's lifecycle.
        InteractorHelper.attach(interactor, presenter, router, null);
        InteractorHelper.detach(interactor);

        throw new RuntimeException("Remove this test and add real tests.");
    }

}

File: app/src/test/java/com/teamdecano/cryptocoin/root/RootInteractorTest.kt
package com.teamdecano.cryptocoin.root

import com.uber.rib.core.InteractorHelper
import com.uber.rib.core.RibTestBasePlaceholder
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.mockito.MockitoAnnotations

class RootInteractorTest : RibTestBasePlaceholder() {

    @Mock internal lateinit var presenter: RootInteractor.RootPresenter
    @Mock internal lateinit var router: RootRouter

    private lateinit var interactor: RootInteractor

    @Before
    fun setup() {
        MockitoAnnotations.initMocks(this)
        interactor = TestRootInteractor.create(presenter)
    }

    @Test
    fun whenBecomeActiveShouldAttachBottomNavigation() {

        InteractorHelper.attach(interactor, presenter, router, null)
        verify(router).attachNavigation()
        InteractorHelper.detach(interactor)
    }
}